disperser/apiserver/disperse_blob_v2.go

```diff
@@ -0,0 +1,126 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api"
+	pb "github.com/Layr-Labs/eigenda/api/grpc/disperser/v2"
+	"github.com/Layr-Labs/eigenda/common"
+	v2 "github.com/Layr-Labs/eigenda/core/v2"
+	dispv2 "github.com/Layr-Labs/eigenda/disperser/common/v2"
+	"github.com/Layr-Labs/eigenda/encoding/rs"
+)
+
+func (s *DispersalServerV2) DisperseBlob(ctx context.Context, req *pb.DisperseBlobRequest) (*pb.DisperseBlobReply, error) {
+	if err := s.validateDispersalRequest(req); err != nil {
+		return nil, err
+	}
+
+	origin, err := common.GetClientAddress(ctx, s.rateConfig.ClientIPHeader, 2, true)
```

Comment:
Does this take into account proxies setting real ip header etc ?

Reply:
Yes, as discussed offline, we want to rate limit against the proxy address if request is routed through a proxy